static/js/pages/instances/actions/StartStopInstanceBtn.tsx

```diff
@@ -2,105 +2,145 @@ import React, { FC, useState } from "react";
         });
         notify.success(`Instance ${instance.name} started.`);
       })
       .catch((e) => {
-        setLoading(false);
+        setStart(false);
         notify.failure("Error on instance start.", e);
       });
   };
 
   const handleStop = () => {
-    setLoading(true);
-    stopInstance(instance)
+    setStop(true);
+    stopInstance(instance, isForce)
       .then(() => {
-        setLoading(false);
+        setStop(false);
         void queryClient.invalidateQueries({
           queryKey: [queryKeys.instances],
         });
         notify.success(`Instance ${instance.name} stopped.`);
       })
       .catch((e) => {
-        setLoading(false);
+        setStop(false);
         notify.failure("Error on instance stop.", e);
       });
   };
 
   return (
     <>
       {canBeStarted && (
-        <ConfirmationButton
-          className={isLoading ? "" : className}
-          isLoading={isLoading}
-          iconClass={
-            isLoading
-              ? "p-icon--spinner u-animation--spin"
-              : "p-icon--video-play"
-          }
-          iconDescription="Start"
-          title="Confirm start"
-          toggleCaption={hasCaption ? "Start" : undefined}
-          confirmationMessage={`Are you sure you want to start instance "${instance.name}"?`}
-          posButtonLabel="Start"
-          onConfirm={handleStart}
-          isDense={isDense}
-          isDisabled={!isEnabled}
-        />
+        <Button
+          hasIcon
+          className={className}
+          dense={isDense}
+          disabled={isDisabled}
+          onClick={handleStart}
+          type="button"
+        >
+          <i
+            className={
+              isStarting
+                ? "p-icon--spinner u-animation--spin"
+                : "p-icon--video-play"
+            }
+          />
+          <span>{isStarting ? "Starting" : "Start"}</span>
+        </Button>
       )}
       {canBeStopped && (
         <ConfirmationButton
-          className={isLoading ? "" : className}
-          isLoading={isLoading}
-          iconClass={
-            isLoading
-              ? "p-icon--spinner u-animation--spin"
-              : "p-icon--power-off"
-          }
+          className={className}
+          isLoading={isStopping}
+          iconClass={"p-icon--power-off"}
           iconDescription="Stop"
           title="Confirm stop"
-          toggleCaption={hasCaption ? "Stop" : undefined}
+          toggleCaption={
+            hasCaption ? (isStopping ? "Stopping" : "Stop") : undefined
+          }
           confirmationMessage={`Are you sure you want to stop instance "${instance.name}"?`}
+          confirmationExtra={
+            <span className="u-float-left">
+              <CheckboxInput
+                inline
+                label="Force stop"
+                defaultChecked={isForce}
+                onClick={() => setForce((prev) => !prev)}
+              />
+            </span>
+          }
```

Comment:
I don't really like the fact that this checkbox gets the focus automatically once the modal gets opened. Is this an intended/desired behaviour?

Reply:
Right, that is a little wired. Good catch.